lib/ui/yt_dlp_form.dart

```diff
@@ -12,95 +13,29 @@ ValueNotifier<MaterialStatesController> downloadButtonNotifier =
     ValueNotifier(MaterialStatesController());
 ValueNotifier<double> downloadPercentageNotifier = ValueNotifier(0);
 
-class YtDlpForm extends StatefulWidget {
+class YtDlpForm extends ConsumerStatefulWidget {
   const YtDlpForm({super.key});
 
   @override
-  State<YtDlpForm> createState() => _YtDlpFormState();
+  ConsumerState<YtDlpForm> createState() => _YtDlpFormState();
 }
 
-class _YtDlpFormState extends State<YtDlpForm> {
-  YtDlpConfig _config = YtDlpConfig.defaultConfig();
-
+class _YtDlpFormState extends ConsumerState<YtDlpForm> {
   String _dlPath = "";
 
-  void setDlAudio(bool? value) {
-    setState(() {
-      _config = _config.set(dlAudio: value);
-    });
-  }
-
-  void setDlVideo(bool? value) {
-    setState(() {
-      _config = _config.set(dlVideo: value);
-    });
-  }
-
-  void setDlThumbnail(bool? value) {
-    _config = _config.set(dlThumbnail: value);
-    setState(() {});
-  }
-
-  void setDlSubtitles(bool? value) {
-    setState(() {
-      _config = _config.set(dlSubtitles: value);
-    });
-  }
-
-  void setYtUrl(String? value) {
-    if (value != _config.ytUrl) {
-      debugPrint("setYtUrl: $value : ${_config.ytUrl}");
-      _config = _config.set(ytUrl: value);
-    }
-  }
-
-  void setVideoSize(VideoSize? value) {
-    setState(() {
-      _config = _config.set(vSize: value);
-    });
-  }
-
-  void setAudioBitrate(AudioBitrate? value) {
-    setState(() {
-      _config = _config.set(aBitrate: value);
-    });
-  }
-
-  void setVideoFormat(VideoFormat? value) {
-    setState(() {
-      _config = _config.set(vFormat: value);
-    });
-  }
-
-  void setAudioFormat(AudioFormat? value) {
-    setState(() {
-      _config = _config.set(aFormat: value);
-    });
-  }
-
-  void setStartTime(String? value) {
-    setState(() {
-      _config = _config.set(startTime: value);
-    });
-  }
-
-  void setEndTime(String? value) {
-    setState(() {
-      _config = _config.set(endTime: value);
-    });
-  }
-
   @override
   Widget build(BuildContext context) {
+    final configApi = ref.watch(ytDlpConfigProvider.notifier);
```

Comment:
It should be ref.read(ytDlpConfigProvider.notifier) , not watch inside build.

Reply:
Using `read` or `watch` won't make any difference here since we are just getting references to immutable functions in the notifier. The actual state is `ytDlpConfigProvider` itself.

Reply:
Please refer to official docs [docs v2](https://docs-v2.riverpod.dev/docs/concepts/reading#dont-use-refread-inside-the-build-method). This is a bad practice as reading notifier should not be used inside build.

Reply:
well, the docs says this:
```
Using ref.read should be avoided as much as possible because it is not reactive.

It exists for cases where using watch or listen would cause issues. If you can, it is almost always better to use watch/listen, especially watch.
```
Also, please take a look at the example where the author used `ref.watch(counterProvider.notifier)` instead of `ref.read(counterProvider.notifier)`

Reply:
but reactiveness when you want to use the value the provider hold(on invalidate or refresh ).but here we want to update some state using notifier.


and doc refers also 

The watch method should not be called asynchronously, like inside an onPressed of an [ElevatedButton](https://api.flutter.dev/flutter/material/ElevatedButton-class.html). Nor should it be used inside initState and other [State](https://api.flutter.dev/flutter/widgets/State-class.html) life-cycles.

In those cases, consider using ref.read instead.

Reply:
Not using watch inside build is a good point